### DISCUSSION TITLE
baseDir configuration when running renovate via Docker

### DISCUSSION THREAD

**Discussion by wwuck** [renovatebot/renovate#11519] [Request Help] [answered]
I couldn't find any documentation or discussions relating to this.  I did find https://github.com/renovatebot/renovate/issues/10526 and https://github.com/renovatebot/renovate/pull/10618 which may be relevant.

What is the current recommended configuration for `baseDir` when running renovate via Docker?  Do I need to mount `baseDir` into the host filesystem or is it mounted automatically elsewhere?  My use case is just to know whether the renovate Docker image will contain enough empty space for cloning/caches/etc. if I leave `baseDir` as default which will be autogenerated in `/tmp/renovate` (I'm guessing inside the Docker container).

**Comment by viceice** [comment, marked as answer]
It depends.
If you use the full image, you don't need to do anything, as all stuff will be done in container.
If you are using the slim image with `binarySource=docker`  you need to mount the docker socket to renovate and the base dir needs to be a host dir too, as renovate will mount parts of `baseDir` and `cacheDir` to the sidecontainers.

On which platform you will run renovate? We have plenty of samples for different ci pipelines, kubernetes charts ....

**Reply by wwuck** [reply]
I am using the full image, as I didn't want the complications of mounting ssh keys into the sidecontainers.  I guess I can just leave `baseDir` out of my config files to use the default values.

**Reply by viceice** [reply]
btw: we are prepareing a FAQ for mounting ssh agent sock with docker sidecar 😉 

https://github.com/renovatebot/renovate/issues/5616#issuecomment-908230845

**Reply by wwuck** [reply]
My current setup for the full image is to have a local ssh directory `chown 1000:1000` and `chmod 0700` and mount `$PWD/ssh:/home/ubuntu/.ssh:ro`.  Would it be any different with the docker sidecars?  Is that FAQ referring to running the ssh-agent inside the renovate container, or on the docker host?

I'll wait for the result of that discussion.

**Reply by viceice** [reply]
The agent will run in host, but it depends how you are starting renovate.

If you are runing the docker container from gitlab pipeline or some other ci, you will run the agent in the renovate container.

 The relevant parts are the socket inside the shared `cacheDir`  and the `customEnvVariables` config.

**Reply by wwuck** [reply]
I was planning on spinning up a new VM running docker, but I guess Jenkins is just as good. Thanks for the tip.  I could then run renovate on the Jenkinsfile pipeline to keep the renovate docker tag updated!

**Reply by viceice** [reply]
there are some jenkins samples here in discussions, just search for it 😉

**Reply by wwuck** [reply]
Thanks.  Found one at https://github.com/renovatebot/renovate/discussions/10762.  It doesn't mention ssh-agent, but I'm sure I'll find plenty about that on the internet searching.  Does the slim renovate image come with ssh-agent pre-installed?
